src/main/kotlin/org/rust/ide/inspections/RsWrongTypeArgumentsNumberInspection.kt

```diff
@@ -30,33 +29,28 @@ class RsWrongTypeParametersNumberInspection : RsLocalInspectionTool() {
 
     private fun checkMethod(holder: RsProblemsHolder, o: RsElement) {
         val (actualArguments, declaration) = when (o) {
-            is RsMethodCall ->
-                o.typeArgumentList to o.reference.resolve()
-
+            is RsMethodCall -> o.typeArgumentList to o.reference.resolve()
             is RsCallExpr ->
                 (o.expr as? RsPathExpr)?.path?.typeArgumentList to (o.expr as? RsPathExpr)?.path?.reference?.resolve()
-
-            is RsBaseType ->
-                o.path?.typeArgumentList to o.path?.reference?.resolve()
-
+            is RsBaseType -> o.path?.typeArgumentList to o.path?.reference?.resolve()
             else -> return
         }
         if (declaration !is RsGenericDeclaration) return
         val nArguments = actualArguments?.typeReferenceList?.size ?: 0
 
-        val expectedRequiredParams = declaration.typeParameterList?.typeParameterList?.filter { it.typeReference == null }?.size ?: 0
+        val expectedRequiredParams = declaration.typeParameterList?.typeParameterList?.filter { it.typeReference == null }?.size
+            ?: 0
         val expectedTotalParams = declaration.typeParameterList?.typeParameterList?.size ?: 0
 
-        val data = when(o) {
+        val data = when (o) {
             is RsBaseType -> checkBaseType(nArguments, expectedRequiredParams, expectedTotalParams)
-            is RsMethodCall -> checkMethodCall(nArguments, expectedRequiredParams, expectedTotalParams)
-            is RsCallExpr -> checkCallExpr(nArguments, expectedRequiredParams, expectedTotalParams)
+            is RsMethodCall, is RsCallExpr -> checkFunctionCall(nArguments, expectedRequiredParams, expectedTotalParams)
             else -> null
         } ?: return
 
-        val problemText = "Wrong number of type parameters: expected ${data.expectedText}, found $nArguments [${data.code}]"
+        val problemText = "Wrong number of type arguments: expected ${data.expectedText}, found $nArguments [${data.code}]"
```

Comment:
I haven't made a review yet, but maybe it's worth to use `org.rust.lang.utils.RsDiagnostic` instead of legacy way when we manually inserted error code without link to rust error index?

Reply:
This is indeed nicer, I switched to `RsDiagnostic`.